main.go

```diff
@@ -176,28 +177,32 @@ func main() {
 
 	logger.Info("starting UI...")
 
-	if !*noUI {
-		go func() {
-			<-done
-			exitErr(errors.New("exit with signal"))
-		}()
+	if *noUI {
+		<-done
+		return
+	}
 
-		if err := setupGUI(privateKey, messenger, logger); err != nil {
-			exitErr(err)
-		}
+	go func() {
+		<-done
+		exitErr(errors.New("exit with signal"))
+	}()
 
-		if err := messenger.Init(); err != nil {
-			exitErr(err)
-		}
+	if err := setupGUI(privateKey, messenger, logger); err != nil {
+		exitErr(err)
+	}
 
-		if err := g.MainLoop(); err != nil && err != gocui.ErrQuit {
-			exitErr(err)
-		}
+	if err := messenger.Init(); err != nil {
+		exitErr(err)
+	}
 
-		g.Close()
-	} else {
-		<-done
+	cancelPolling := make(chan struct{}, 1)
+	startPolling(g, pollFunc, 50*time.Millisecond, cancelPolling)
+	defer close(cancelPolling)
+	if err := g.MainLoop(); err != nil && err != gocui.ErrQuit {
+		close(cancelPolling)
```

Comment:
This is closed twice. Here and `defer close(cancelPolling)`.

Reply:
There's no problem in closing a channel twice, right? The reason I did it was to make sure we cancel polling before closing the GUI.

Reply:
It is, it will panic. However, in this case, it will exit before calling the `defer close(cancelPolling)`. I think closing this channel in `defer` is enough.

I would refactor this fragment if you really want to keep the order of actions:
```go
cancelPolling := make(chan struct{}, 1)
startPolling(g, pollFunc, 50*time.Millisecond, cancelPolling)
err := g.MainLoop()
close(cancelPolling) // stop polling before the GUI loop is stopped regardless of the error
if err != nil && err != gocui.ErrQuit {
    exitErr(err)
}
g.Close()
```